client/src/components/AddProduct.js

```diff
@@ -22,15 +24,67 @@ const AddProduct = (props) => {
     changeListOpen,
     addProductOpen,
     changeAddProductOpen,
-    demoListsArray,
+    lists,
   } = props;
   const classes = useStyles();
-  const [url, setUrl] = useState("");
+  const [inputLink, setInputLink] = useState("");
   const [list, setList] = useState("");
   const [errors, setErrors] = useState({});
+  const userId = useContext(UserContext).userId;
+  const [item, setItem] = useState({});
+  const [selectedListIndex, setSelectedListIndex] = useState(0);
+  const [listId, setListId] = useState("");
```

Comment:
Do we need list, listId, and selectedListIndex? Can we store just the index and use that to access the list and listId when needed?

Reply:
Quite possibly, it could definitely be a lot cleaner in general, I left that for later because I wanted to get something functioning and good to go on the dev branch, but there is some cleanup in general and some spare parts left over from various testing and trying to get it to work that should be taken out.